### PR TITLE
datasets: fix memuse to include string len

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/3910

Note: This could be a breaking change for existing setups as indicated in the discussion on the redmine ticket.

### CODE REVIEW

**Review by jufajardini — COMMENTED** [review]
As this is still in draft state, I sense you may have planned for this, but just so we won't forget: if this could break existing setups, we should update the upgrading docs section :)

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 20550

File: src/util-thash.c [hunk 1]
```diff
@@ -334,6 +334,11 @@ THashTableContext *THashInit(const char *cnf_prefix, size_t data_size,
     return ctx;
 }
 
+void THashUpdateMemuse(THashTableContext *ctx, const size_t data_size)
```
Comment (jasonish) [inline src/util-thash.c:337]: Should use a name that suggests adding or incrementing here.  Probably also need to track decrementing as well.

Comment (inashivb) [reply src/util-thash.c:337]: Indeed. That's a big mistake.

**Pull request closed without merging**